test/xUnit/csharp/test_Prediction.cs

```diff
@@ -0,0 +1,190 @@
+
+        public static readonly MyPredictor SlowPredictor;
+
+        public static readonly MyPredictor FastPredictor;
+
+        static MyPredictor()
+        {
+            SlowPredictor = new MyPredictor(
+                Guid.NewGuid(),
+                "Test Predictor #1",
+                "Description for #1 predictor.",
+                delay: true);
+
+            FastPredictor = new MyPredictor(
+                Guid.NewGuid(),
+                "Test Predictor #2",
+                "Description for #2 predictor.",
+                delay: false);
+        }
+
+        private MyPredictor(Guid id, string name, string description, bool delay)
+        {
+            _id = id;
+            _name = name;
+            _description = description;
+            _delay = delay;
+
+            History = new List<string>();
+            AcceptedSuggestions = new List<string>();
+        }
+
+        public Guid Id => _id;
+
+        public string Name => _name;
+
+        public string Description => _description;
+
+        bool IPredictor.SupportEarlyProcessing => true;
+
+        bool IPredictor.AcceptFeedback => true;
+
+        public void StartEarlyProcessing(IReadOnlyList<string> history)
+        {
+            History.AddRange(history);
+        }
+
+        public void OnSuggestionAccepted(string acceptedSuggestion)
+        {
+            AcceptedSuggestions.Add(acceptedSuggestion);
+        }
+
+        public List<PredictiveSuggestion> GetSuggestion(PredictionContext context, CancellationToken cancellationToken)
+        {
+            if (_delay)
+            {
+                // The delay is exaggerated to make the test reliable.
+                // xUnit must spin up a lot tasks, which makes the test unreliable when the time difference between 'delay' and 'timeout' is small.
+                Thread.Sleep(2000);
+            }
+
+            // You can get the user input from the AST.
+            var userInput = context.InputAst.Extent.Text;
+            return new List<PredictiveSuggestion> {
+                new PredictiveSuggestion($"{userInput} TEST-1 from {Name}"),
+                new PredictiveSuggestion($"{userInput} TeSt-2 from {Name}"),
+            };
+        }
+    }
+
+    public static class CommandPredictionTests
+    {
+        [Fact]
+        public static void PredictInput()
+        {
+            MyPredictor slow = MyPredictor.SlowPredictor;
+            MyPredictor fast = MyPredictor.FastPredictor;
+            Ast ast = Parser.ParseInput("Hello world", out Token[] tokens, out _);
+
+            // Returns null when no predictor implementation registered
+            List<PredictionResult> results = CommandPrediction.PredictInput(ast, tokens).Result;
+            Assert.Null(results);
+
+            try
+            {
+                // Register 2 predictor implementations
+                SubsystemManager.RegisterSubsystem<IPredictor, MyPredictor>(slow);
```

Comment:
Will the follow work too:
```suggestion
                SubsystemManager.RegisterSubsystem(slow);
```

Reply:
Answered above.